stablehlo/dialect/StablehloOps.cpp

```diff
@@ -130,6 +130,20 @@ LogicalResult ReduceScatterOp::verify() {
       channelId, getUseGlobalDeviceIds(), getComputation(), getResult());
 }
 
+mlir::Speculation::Speculatability ReduceScatterOp::getSpeculatability() {
+  auto inputType = cast<RankedTensorType>(getOperand().getType());
+  auto resultType = cast<RankedTensorType>(getResult().getType());
```

Comment:
Do we still need the `cast`?

```
RankedTensorType inputType = getOperand().getType();
```

Reply:
Good point, we don't! Removed, here and below.

Reply:
There's some others in this file, I'll clean that up soon.